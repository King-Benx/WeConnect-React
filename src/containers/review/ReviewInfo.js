import React from 'react';
import comment_icon from '../../images/comment-icon.jpg';
import { Image } from 'react-bootstrap';

class ReviewInfo extends React.Component {
    constructor(props){
        super(props);
        this.state = {}
    }

    render() { 
        return (
            // JSX returned
            <div key={this.props.review.id} className="row">
                <div className="col-xs-2">
                    <Image src={ comment_icon } alt="" responsive circle/>
                </div>
                <div className="col-xs-10">
                    <h4 className="list-group-item-heading">Author: <span className="text-primary">{ this.props.review.author }</span></h4>
                    <blockquote className="list-group-item-text">{ this.props.review.review }</blockquote>
                    <small className="text text-right text-primary">{ this.props.review.last_modified }</small>
                    <hr/>
                </div>
                <br/>
            </div> 
        )
    }
}
 
export default ReviewInfo;