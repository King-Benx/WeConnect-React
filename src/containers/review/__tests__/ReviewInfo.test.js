import React from 'react';
import { shallow } from 'enzyme';
import ReviewInfo from "../ReviewInfo";
import renderer from 'react-test-renderer';

describe('These are tests for the login form ', () =>{
    const props = {
        match: {
            params: {
            }
        },
        review:{
        }
    }
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }

    let mountedReviewInfo;

    beforeEach(()=>{
        mountedReviewInfo = shallow(<ReviewInfo {...props}/>);
    });

    it('renders without crashing', () => {
       shallow(<ReviewInfo {...props} />)
       const tree = renderer.create(<ReviewInfo {...props} />).toJSON();
       expect(tree).toMatchSnapshot();
    });

    it('has div', () => {
        const divs = mountedReviewInfo.find('div')
        expect(divs.length).toBe(3)
    });

    it('has a blockquote', () => {
        const blockquotes = mountedReviewInfo.find('blockquote')
        expect(blockquotes.length).toBe(1)
    });

    it('has small size paragraphs', () => {
        const smalls = mountedReviewInfo.find('small')
        expect(smalls.length).toBe(1)
    });

    it('has an hr', () => {
        const hrs = mountedReviewInfo.find('hr')
        expect(hrs.length).toBe(1)
    });

    it('has an br', () => {
        const brs = mountedReviewInfo.find('br')
        expect(brs.length).toBe(1)
    });

    it('has an h4', () => {
        const h4s = mountedReviewInfo.find('h4')
        expect(h4s.length).toBe(1)
    });

    it('has an image', () => {
        const images = mountedReviewInfo.find('Image')
        expect(images.length).toBe(1)
    });

});