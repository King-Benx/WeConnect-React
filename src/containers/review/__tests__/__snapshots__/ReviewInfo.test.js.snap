// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the login form  renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-2"
  >
    <img
      alt=""
      className="img-responsive img-circle"
      src="comment-icon.jpg"
    />
  </div>
  <div
    className="col-xs-10"
  >
    <h4
      className="list-group-item-heading"
    >
      Author: 
      <span
        className="text-primary"
      />
    </h4>
    <blockquote
      className="list-group-item-text"
    />
    <small
      className="text text-right text-primary"
    />
    <hr />
  </div>
  <br />
</div>
`;
